File: mfi_customization/mfi/web_form/issues/issues.js

frappe.ready(function() {


frappe.web_form.after_load = () =>{


     frappe.web_form.set_value("opening_date_time",frappe.datetime.now_datetime())
    
	    
frappe.call({

method: 'mfi_customization.mfi.web_form.issues.issues.get_logged_user',
    callback: function(r) {
        console.log("r.message", r.message)
        if ((r.message).length == 1){
            frappe.web_form.set_value("customer",r.message)
        }
        else if ((r.message).length > 1){
            var options =[]
            $.each(r.message || [], function(i, d) {
                options.push({
                 'label':d,
                 'value':d
                 
                });
            });
            var field = frappe.web_form.get_field("customer")
            field._data = options;
            field.refresh();
        }
    }
})            


frappe.web_form.on('customer', () => {
  

		frappe.call({
    method: 'mfi_customization.mfi.web_form.issues.issues.get_Customer_name',
    args: {
        customer:frappe.web_form.get_value("customer")
    },
    async: true,
    callback: function(r) {

      frappe.web_form.set_value("name_of_the_customer",r.message)
        
    }
});

})

frappe.web_form.on('name_of_the_customer', () => {

location()
assetfunction()
serialnno()
})

frappe.web_form.on('location', () => {
assetfunction()
serialnno()
})

frappe.web_form.on('asset', () => {

//location()
get_location_serialnoby_asset()

})

frappe.web_form.on('serial_no', () => {

//location()
get_location_assetByseriaNo()

})

function location(){
		frappe.call({
    method: 'mfi_customization.mfi.web_form.issues.issues.get_location',
    args: {
        customerId:frappe.web_form.get_value("customer")
    },
      
    callback: function(r) {
        var options =[]
        
        for (const [key,value] of Object.entries(r.message)) {
            
         options.push({
         'label':value.location,
         'value':value.location
         
         });
   
   
        }

        var field = frappe.web_form.get_field("location")
         field._data = options;
         field.refresh();
        
    }
});




}

function assetfunction(){

		frappe.call({
    method: 'mfi_customization.mfi.web_form.issues.issues.get_Asset',
    args: {
        customerId:frappe.web_form.get_value("customer"),
        location:frappe.web_form.get_value("location")
    },
      
    callback: function(r) {
        var options =[]
        if (frappe.web_form.get_value("location")==""){
        
                for (const [key,value] of Object.entries(r.message[1])) {
         
            
         options.push({
         'label':value.name,
         'value':value.name
         
         });
   
   
        }
        
        
        }
        if (frappe.web_form.get_value("location")!=""){
        
               for (const [key,value] of Object.entries(r.message[1])) {     
         options.push({
         'label':value.name,
         'value':value.name
         
         });
   
   
        }
        
        }
        

     else {
        for (const [key,value] of Object.entries(r.message[0])) {   
         options.push({
         'label':value.name,
         'value':value.name
         
         });
   
  
        }
        }
        var field = frappe.web_form.get_field("asset")
         field._data = options;
         field.refresh();
        
        
    }
});


}

function serialnno(){

frappe.call({
     method: 'mfi_customization.mfi.web_form.issues.issues.get_serialNo',
     args: {
         customerId:frappe.web_form.get_value("customer"),
         location:frappe.web_form.get_value("location")
        },
  
  callback: function(r) {
      
        var options =[]
     if(frappe.web_form.get_value("location")==""){
           
          for (const [key,value] of Object.entries(r.message[0])) {
          
              options.push({
                'label':value.serial_no,
                 'value':value.serial_no
                 
                  });
             }
   
       }
        
     if(frappe.web_form.get_value("location")!=""){
         for (const [key,value] of Object.entries(r.message[0])) {
            options.push({
               'label':value.serial_no,
               'value':value.serial_no
             });
   
         }
    }
        
    else  {
         for (const [key,value] of Object.entries(r.message[1])) {
           options.push({
              'label':value.serial_no,
              'value':value.serial_no
               });
        }
    }
       
        var field = frappe.web_form.get_field("serial_no")
         field._data = options;
         field.refresh();   
      }
   });

}


function get_location_serialnoby_asset(){
        frappe.call({
            method: 'mfi_customization.mfi.web_form.issues.issues.get_location_serialnoby_asset',
      args: {
        asset:frappe.web_form.get_value("asset")

        },
       callback: function(r){
        
         for (let t of Object.values(r.message)){
   
          if(frappe.web_form.get_value("asset")){
           frappe.web_form.set_value("location",t.location)
           frappe.web_form.set_value("serial_no",t.serial_no)
           frappe.web_form.set_value("asset_name",t.asset_name)
           
             }
         
       }
        
    }
        
 });
}

function get_location_assetByseriaNo(){
        frappe.call({
            method: 'mfi_customization.mfi.web_form.issues.issues.get_location_assetByseriaNo',
      args: {
        serialno:frappe.web_form.get_value("serial_no")

        },
       callback: function(r){
       
       for (let i of Object.values(r.message)){
            if(frappe.web_form.get_value("serial_no")){
        
              frappe.web_form.set_value("location",i.location)
              frappe.web_form.set_value("asset",i.name)
            }
        }
    
     }
  });



 }
 
}

});

File: mfi_customization/mfi/web_form/issues/issues.py
from __future__ import unicode_literals

import frappe

def get_context(context):
	# do your magic here
	pass
	
	
@frappe.whitelist(allow_guest=True)
def get_logged_user():
    #user = frappe.db.get_value('User',{"name":frappe.session.user},"full_name")
    # Project_name=frappe.db.get_value("User Permission",{"user":frappe.session.user,"allow":"Project"},"for_value")
    customer_list =[u.get('for_value') for u in frappe.db.get_list("User Permission",{"user":frappe.session.user,"allow":"Customer"},"for_value")]
    print("customer_list",customer_list)
    # customerId=frappe.db.get_value("Project",{"name":Project_name},"customer")
   
    return customer_list  

    frappe.response["message"] = {
        "success_key":1,
        "message":"Authentication success",
        "sid":frappe.session.sid,
        "api_key":user.api_key,
        "username":user.username,
        "email":user.email
    }
	

@frappe.whitelist()		
def get_location(customerId):
    location_list=[]
    project_list = [p.name for p in frappe.db.get_list("Project", filters={"customer":customerId},fields={"name"})]
    for p in project_list:
        location =frappe.db.get_list("Asset",{"project":p},"location")
        [location_list.append(Location) for Location in location if Location not in location_list]
    return location_list   
        
	
@frappe.whitelist()		
def get_Customer_name(customer):
    customer_name=frappe.db.get_value("Customer",{"name":customer},"customer_name")
    return customer_name



@frappe.whitelist()
def get_Asset(customerId,location):
    on_basis_company_customer=[]
    cpny_and_locn1=[]
    project_list = [p.name for p in frappe.db.get_list("Project", filters={"customer":customerId},fields={"name"})]
    for p in project_list:
        company =frappe.db.get_list("Asset",filters={"project":p},fields={"company"})
        
        for i in company:
            company_customer =frappe.db.get_all("Asset",filters={"company":i.company,"project":p},fields={"name"})
            [on_basis_company_customer.append(value) for value in company_customer if value not in on_basis_company_customer ]
            cpny_and_locn=frappe.db.get_all("Asset",filters={"company":i.company,"project":p,              "location":location},fields={"name"})
            [cpny_and_locn1.append(value1) for value1 in cpny_and_locn if value1 not in cpny_and_locn1 ]
    return on_basis_company_customer,cpny_and_locn1

   	
@frappe.whitelist()   	
def get_serialNo(customerId,location):
    bycustomer_location =[]
    bycustomer =[]
    project_list = [p.name for p in frappe.db.get_list("Project", filters={"customer":customerId},fields={"name"})]
    for p in project_list:
        locn_cust=frappe.db.get_all("Asset",filters={"project":p,"location":location},fields={"serial_no"})
        [bycustomer_location.append(i) for i in locn_cust if i not in bycustomer_location]
        ByCustomerFilter= frappe.db.get_all("Asset",filters={"project":p},fields={"serial_no"})
        [bycustomer.append(j) for j  in ByCustomerFilter if j not in bycustomer]
    return bycustomer_location,bycustomer
    
    
     
     
@frappe.whitelist()     
def get_location_serialnoby_asset(asset):
    location_serial=frappe.db.get_list("Asset",filters={"name":asset},fields={"location","serial_no","asset_name"})
    return location_serial      
     
     
@frappe.whitelist()     
def get_location_assetByseriaNo(serialno):
    location_asset=frappe.db.get_list("Asset",filters={"serial_no":serialno},fields={"location","name"})
    return location_asset
         
     
     
     
     
 
    	
